content/influxdb/cloud-dedicated/query-data/troubleshoot.md

```diff
@@ -0,0 +1,194 @@
+  Learn how to handle responses and troubleshoot errors encountered when querying {{% cloud-name %}} with Flight+gRPC and Arrow Flight clients.
+weight: 201
+menu:
+  influxdb_cloud_dedicated:
+    name: Troubleshoot responses and errors
+    parent: Query data
+influxdb/cloud-dedicated/tags: [query, sql, influxql]
+alt_engine:
+---
+
+Learn how to handle responses and troubleshoot errors encountered when querying {{% cloud-name %}} with Flight+gRPC and Arrow Flight clients.
+
+<!-- TOC -->
+
+- [InfluxDB Flight responses](#influxdb-flight-responses)
+  - [Schema](#schema)
+    - [Example](#example)
+  - [InfluxDB error codes](#influxdb-error-codes)
+  - [Troubleshoot errors](#troubleshoot-errors)
+    - [Internal Error: Received RST_STREAM](#internal-error-received-rst_stream)
+    - [Internal Error: stream terminated by RST_STREAM with NO_ERROR](#internal-error-stream-terminated-by-rst_stream-with-no_error)
+    - [Invalid Argument Error: bucket <BUCKET_ID> not found](#invalid-argument-error-bucket-bucket_id-not-found)
+      - [Unauthenticated: Unauthenticated](#unauthenticated-unauthenticated)
+    - [Unauthenticated: read:<BUCKET_ID> is unauthorized](#unauthenticated-readbucket_id-is-unauthorized)
+
+## InfluxDB Flight responses
+
+{{% cloud-name %}} provides an InfluxDB-specific Arrow Flight service that uses gRPC Remote Procedure Calls (gRPC) to transport query result data in Arrow format.
+Flight defines a set of [RPC methods](https://arrow.apache.org/docs/format/Flight.html#rpc-methods-and-request-patterns) for servers and clients.
+Using gRPC, a Flight client (for example, `influx3` or an InfluxDB v3 client library) calls an InfluxDB Flight RPC server method to send a request to the server.
+In gRPC, every call returns a status object that contains an integer code and a string message.
+While gRPC defines the integer [status codes](https://grpc.github.io/grpc/core/status_8h.html) and definitions for servers and clients,
+a server can choose which status to return for an RPC call.
+During a request, the gRPC client and server may each return a status independent of the other.
+
+InfluxDB v3 client library query methods implement the Flight [`DoGet(FlightCallOptions, Ticket)` method](https://arrow.apache.org/docs/cpp/api/flight.html#_CPPv4N5arrow6flight12FlightClient5DoGetERK17FlightCallOptionsRK6Ticket) that retrieves a single stream for the the request.
+For example, if you call the `influxdb3-python` Python client library `InfluxDBClient3.query()` method, the client in turn calls the `pyarrow.flight.FlightClient.do_get()` method and passes a Flight ticket with your credentials and query to InfluxDB.
+
+InfluxDB v3 can respond with one of the following:
+
+- The schema for record batches in the stream, followed by a stream of query result data (`RecordBatch`), the request status (`OK`), and optional trailing metadata
+- The schema, a stream or partial stream of data, and no status (for example, in the event of a network interruption)
```

Comment:
I am not sure calling out a truncated stream is particularly valuable as there is nothing influx / flight specific about that. I would expect the client to return an error in this case if the network connection was interrupted